### PR TITLE
Update Typespec for Language Conversations API

### PR DESCRIPTION
**Pull request by quentinRobinson**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

Moving to typsespec. fixed some errors in tsp and updated the LRO operations to be able to generate the SDK

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/cognitiveservices/Language.Conversations/common.tsp [hunk 1]
```diff
@@ -80,7 +88,7 @@ enum State {
 @added(Versions.v2023_04_01)
 model TaskIdentifier {
     @doc("task name")
-    taskName: string;
+    taskName?: string;
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/common.tsp:91]: How can `taskName` be optional when it's the only property?

**Review by heaths — CHANGES_REQUESTED** [review]
Mostly looks fine, but where did you pick up the files outside your service directory? A merge from main? Seeing the same thing in another PR that looks similar.

File: specification/cognitiveservices/Language.Conversations/common.tsp (hunk 1 above)
Comment (quentinRobinson) [reply specification/cognitiveservices/Language.Conversations/common.tsp:91]: That is how we have it defined in the [json spec](https://github.com/Azure/azure-rest-api-specs/blob/4c38cb9966cd6afbb03c7e9b14997720a728baee/specification/cognitiveservices/data-plane/Language/stable/2023-04-01/common.json#L206)

**Comment by heaths** [comment]
See my [comment on the other PR](https://github.com/Azure/azure-rest-api-specs/pull/26175#issuecomment-1786194063).

**Comment by quentinRobinson** [comment]
I am also trying to get our team to review this PR to make sure everything looks right to them as well

**Pull request closed without merging**